### PR TITLE
feature - Included the e2e into the PR workflows

### PR DESCRIPTION
**Pull request by naveensrinivasan**
- Validated the presence of the GITHU_AUTH_TOKEN variable presence before running the e2e.
- Update the contributing doc with scopes of the personal access token.
- Updated the workflow to include the e2e tests.

### Secret 
A secret name `GIT_AUTH_TOKEN` has to be created with Personal access token to run `e2e` before merging this PR.

### CODE REVIEW

File: e2e/e2e_suite_test.go
Comment (inferno-chromium) [inline e2e/e2e_suite_test.go:36]: nit: space after //

**Review by inferno-chromium — APPROVED** [review]

**Review by inferno-chromium — APPROVED** [review]

**Comment by naveensrinivasan** [comment]
@inferno-chromium  This has to be done prior to merging this PR.

>Secret
A secret name GIT_AUTH_TOKEN has to be created with Personal access token to run e2e before merging this PR.

**Comment by naveensrinivasan** [comment]
And that is the reason why the `build` is failing.

**Comment by inferno-chromium** [comment]
> @inferno-chromium This has to be done prior to merging this PR.
> 
> > Secret
> > A secret name GIT_AUTH_TOKEN has to be created with Personal access token to run e2e before merging this PR.

GITHUB_AUTH_TOKEN ? and not GIT_AUTH_TOKEN
Where do we need to configure this, i think you should have write access to project settings, can you create a low privilege token and add it. @dlorenc fyi

**Comment by naveensrinivasan** [comment]
> > @inferno-chromium This has to be done prior to merging this PR.
> > > Secret
> > > A secret name GIT_AUTH_TOKEN has to be created with Personal access token to run e2e before merging this PR.
> 
> GITHUB_AUTH_TOKEN ? and not GIT_AUTH_TOKEN
> Where do we need to configure this, i think you should have write access to project settings, can you create a low privilege token and add it. @dlorenc fyi

Here is where it needs to be created.

image

I don't have permission to do it.

I tried `GITHUB_AUTH_TOKEN` but GitHub rejected me from using that name and that is the reason I had to use `GIT_ATUH_TOKEN`

**Comment by inferno-chromium** [comment]
we will be adding gitlab support in sometime, so maybe lets use GITHUB_TOKEN ? @dlorenc - can you please create a low-privilege token and add it there.

**Comment by inferno-chromium** [comment]
@naveensrinivasan - it wont work since

"With the exception of GITHUB_TOKEN, secrets are not passed to the runner when a workflow is triggered from a forked repository."
https://docs.github.com/en/free-pro-team@latest/actions/reference/encrypted-secrets#using-encrypted-secrets-in-a-workflow

this would be painful for external contributors.

**Comment by naveensrinivasan** [comment]
@inferno-chromium Here is my fork and I am able to pass secrets. 
 I had to create those secrets in my fork https://github.com/naveensrinivasan/scorecard/runs/1670757896?check_suite_focus=true , none of the actions run by default in any of the forks.

**Comment by naveensrinivasan** [comment]
I was able to create a secret `GH_AUTH_TOKEN` and with your push to this PR , the action was successful https://github.com/naveensrinivasan/scorecard/actions/runs/474656737

**Comment by inferno-chromium** [comment]
> I was able to create a secret `GH_AUTH_TOKEN` and with your push to this PR , the action was successful https://github.com/naveensrinivasan/scorecard/actions/runs/474656737

See this why - https://stackoverflow.com/questions/58737785/github-actions-empty-env-secrets
it is returning empty since you do not have write access (you are currently maintainer, but i can get you write access).

I am worried about security implication of switching to pull_request_target. So, how about we do this:
1. Give you write access to repo so that all integration checks (that require secrets/token) runs for admins and you.
2. PRs from other contributors don't fail on this integration check and hence should bail out for this, so you need to convert those checks into bail out conditions.

Thoughts ?

**Comment by dlorenc** [comment]
> 2\. PRs from other contributors don't fail on this integration check and hence should bail out for this, so you need to convert those checks into bail out conditions.

Is there any way in GitHub actions for maintainers to trigger the run on a PR from an untrusted contributor? That's how most projects I've worked on are setup, though none have used GitHub actions. In k8s and other setups, maintainers add an /ok-to-test comment or label before the tests can be run.

**Comment by naveensrinivasan** [comment]
> > 2. PRs from other contributors don't fail on this integration check and hence should bail out for this, so you need to convert those checks into bail out conditions.
> 
> Is there any way in GitHub actions for maintainers to trigger the run on a PR from an untrusted contributor? That's how most projects I've worked on are setup, though none have used GitHub actions. In k8s and other setups, maintainers add an /ok-to-test comment or label before the tests can be run.

There are a couple of options based on this IMO

-  https://github.com/jpmcb/prow-github-actions 
     - ☝️ Do know the reliability of this GitHub Action
- Deploy https://github.com/kubernetes/test-infra/tree/master/prow
     -  ☝️ Lot of reliability, but comes with complexity. This would be a great addition if `OSSF` could extend it to other repositories.

**Comment by naveensrinivasan** [comment]
AFAIK this should to suffice to this repo needs https://github.com/jpmcb/prow-github-actions/blob/main/docs/commands.md

**Comment by inferno-chromium** [comment]
> AFAIK this should suffice to this repo needs https://github.com/jpmcb/prow-github-actions/blob/main/docs/commands.md
> But it does not have `ok to test`

What are the next steps here ? I have given you write access, so this PR will merge if you reruns checks, but we need to think/fix if we get external contributions.

**Comment by naveensrinivasan** [comment]
It still didn't work. 

We could close this PR and implement https://github.com/kubernetes/test-infra/tree/master/prow?

Thoughts on it?

**Comment by naveensrinivasan** [comment]
@dlorenc @inferno-chromium  Any thoughts?

**Comment by inferno-chromium** [comment]
> @dlorenc @inferno-chromium Any thoughts?

retriggered checks. i am fine with merging this since we are the only devs right now. but need to find some time for prow, i dont think i have time right now. also i feel there are other priorities first, e.g. adding more checks and adding gitlab api support.

**Comment by naveensrinivasan** [comment]
> > @dlorenc @inferno-chromium Any thoughts?
> 
> retriggered checks. i am fine with merging this since we are the only devs right now. but need to find some time for prow, i dont think i have time right now. also i feel there are other priorities first, e.g. adding more checks and adding gitlab api support.

Sounds good! Thanks. It is still failing.

**Comment by inferno-chromium** [comment]
> > > @dlorenc @inferno-chromium Any thoughts?
> > 
> > 
> > retriggered checks. i am fine with merging this since we are the only devs right now. but need to find some time for prow, i dont think i have time right now. also i feel there are other priorities first, e.g. adding more checks and adding gitlab api support.
> 
> Sounds good! Thanks. It is still failing.

Can you upload another PR, from a branch of this repo itself, without using fork (you have write, dont need to fork).

**Comment by naveensrinivasan** [comment]
closing this  because it has been addressed by #131

**Pull request closed without merging**
